### PR TITLE
src/authCloud: add oauth handler for icarus

### PR DESCRIPTION
**Pull request by ibihim**

### CODE REVIEW

File: example/server.js [hunk 1]
```diff
@@ -1,10 +1,20 @@
...
-let port = 8888;
+const express = require('express');
+const cors = require('cors');
+const proxy = require('http-proxy-middleware');
+const app = express();
+const host = '0.0.0.0';
+const port = 8888;
 
 app.use('/', express.static('./'));
+app.use('/icarus', proxy({
```
Comment (Harsimran1) [inline example/server.js]: I think you can skip this proxy step by setting icarus url to be `http://127.0.0.1:8082` for local development.

Comment (ibihim) [reply example/server.js]: nope, I tried it. didn't work :cry:

File: test/auth/indexTest.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    },
+
+    createCAP() {
+        return Promise.resolve(capObj);
+    },
+});
+
+const tokenResponse = { privateKey: 'secret', token: 'i-am-an-access-token' };
+
+describe.skip('auth', () => {
```
Comment (ibihim) [inline test/auth/indexTest.js]: I will not touch this right now, but I hope to continue adding tests tomorrow on...

File: src/authCloud/index.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+            return false;
+        });
+    },
+
+    // Object -> _
+    config({ clientId, clientURL, gcSDK }) {
+        const sdk = gcSDK || window.GC.SDK;
+        // making the CAP proactively
+        this.cap = sdk.createCAP();
```
Comment (Glumli) [inline src/authCloud/index.js]: this.cap

**Pull request closed without merging**